packages/uicore/src/components/Select/Select.tsx

```diff
@@ -202,16 +203,27 @@ export function Select(props: SelectProps): React.ReactElement {
           {items.filter(item => item.label.toString().toLowerCase().includes(query.toLowerCase())).length === 0 ? (
             <div className={css.noResultsFound}>No Match Found</div>
           ) : null}
-          {props.allowCreatingNewItems && !isExactQueryElPresent && (
-            <Button
-              intent="primary"
-              minimal
-              text={query}
-              icon="plus"
-              className={css.createNewItemButton}
-              onClick={handleClick as React.MouseEventHandler<Element>}
-            />
-          )}
+          {props.allowCreatingNewItems && !isExactQueryElPresent ? (
```

Comment:
Instead of passing the string can we have a react element instead as customRenderer which will help to support any future use cases as well ?